File: test/fixtures/go/small/entity_service_100.go
// Go Fixture (target ~100 lines)
// Generated for benchmark testing

package service

import (
    "context"
    "time"
)

type Entity struct {
    ID        uint64    `json:"id"`
    Name      string    `json:"name"`
    Email     string    `json:"email"`
    IsActive  bool      `json:"is_active"`
    CreatedAt time.Time `json:"created_at"`
    UpdatedAt time.Time `json:"updated_at"`
}

type CreateDto struct {
    Name  string `json:"name"`
    Email string `json:"email"`
}

type UpdateDto struct {
    Name     *string `json:"name,omitempty"`
    Email    *string `json:"email,omitempty"`
    IsActive *bool   `json:"is_active,omitempty"`
}

type EntityService struct {
    db     *Database
    logger *Logger
    cache  *Cache
}

func NewEntityService(db *Database, logger *Logger, cache *Cache) *EntityService {
    return &EntityService{
        db:     db,
        logger: logger,
        cache:  cache,
    }
}


func (s *EntityService) Operation0(ctx context.Context, id uint64, data string) (*Entity, error) {
    result, err := s.db.Query(ctx, "SELECT * FROM entities WHERE id = $1", id)
    if err != nil {
        s.logger.Error("Operation failed", "error", err)
        return nil, err
    }
    s.logger.Debug("Fetched entity", "id", id)
    return parseEntity(result), nil
}

func (s *EntityService) Operation1(ctx context.Context, id uint64, data string) (*Entity, error) {
    result, err := s.db.Query(ctx, "SELECT * FROM entities WHERE id = $1", id)
    if err != nil {
        s.logger.Error("Operation failed", "error", err)
        return nil, err
    }
    s.logger.Debug("Fetched entity", "id", id)
    return parseEntity(result), nil
}

func (s *EntityService) Operation2(ctx context.Context, id uint64, data string) (*Entity, error) {
    result, err := s.db.Query(ctx, "SELECT * FROM entities WHERE id = $1", id)
    if err != nil {
        s.logger.Error("Operation failed", "error", err)
        return nil, err
    }
    s.logger.Debug("Fetched entity", "id", id)
    return parseEntity(result), nil
}

func (s *EntityService) Operation3(ctx context.Context, id uint64, data string) (*Entity, error) {
    result, err := s.db.Query(ctx, "SELECT * FROM entities WHERE id = $1", id)
    if err != nil {
        s.logger.Error("Operation failed", "error", err)
        return nil, err
    }
    s.logger.Debug("Fetched entity", "id", id)
    return parseEntity(result), nil
}

File: test/fixtures/typescript/small/entity_service_100.ts
// TypeScript Fixture (target ~100 lines)
// Generated for benchmark testing

import { Database } from './database';
import { Logger } from './logger';
import { Cache } from './cache';

interface Entity {
    id: number;
    name: string;
    email: string;
    isActive: boolean;
    createdAt: Date;
    updatedAt: Date;
}

interface CreateDto {
    name: string;
    email: string;
}

interface UpdateDto {
    name?: string;
    email?: string;
    isActive?: boolean;
}

type EntityResponse = Promise<Entity | null>;
type EntitiesResponse = Promise<Entity[]>;

class EntityService {
    private db: Database;
    private logger: Logger;
    private cache: Cache<number, Entity>;

    constructor(database: Database, logger: Logger, cache: Cache<number, Entity>) {
        this.db = database;
        this.logger = logger;
        this.cache = cache;
    }


    async operation0(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation0`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation1(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation1`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

    async operation2(id: number, data: string): Promise<Entity> {
        try {
            const result = await this.db.query<Entity>(
                'SELECT * FROM entities WHERE id = ?',
                [id]
            );
            this.logger.debug(`Fetched operation2`);
            return result;
        } catch (error) {
            this.logger.error('Operation failed:', error);
            throw error;
        }
    }

}

export { EntityService, Entity, CreateDto, UpdateDto };
